lib/distrobaker.py

```diff
@@ -796,7 +797,15 @@ def get_buildsys(which):
     if which not in ('source', 'destination'):
         logger.error('Cannot get "%s" build system.', which)
         return None
-    if not hasattr(get_buildsys, which):
+
+    session_timed_out = False
+    if hasattr(get_buildsys, which):
+        session_age = datetime.datetime.now() - getattr(get_buildsys, which)['session_start_time']
+        #slightly less than an hour, to be safe
+        if session_age.seconds > 3550 or session_age.days > 0:
+            session_timed_out = True
```

Comment:
In the case of a timeout, would it be appropriate to attempt to `logout()` the old koji session before establishing a new one below?

Reply:
Thanks, added.